apps/ui/src/components/AssetSelector/GroupedAssetList.tsx

```diff
@@ -19,25 +32,146 @@ function groupBy<A>(group: Group<A>, assets: A[]): GroupedAsset<A> {
   }, {});
 }
 
+enum SearchStatus {
+  None,
+  Invalid,
+  Unregistered,
+}
+
+const SearchMeta = ({ status, chainType }: { status: SearchStatus; chainType: ChainType }) => {
+  switch (status) {
+    case SearchStatus.None:
+      return null;
+    case SearchStatus.Invalid:
+      return (
+        <MetaContainer>{i18n.t(`validation.${chainType === 'Ethereum' ? 'address' : 'type-hash'}`)}</MetaContainer>
+      );
+    case SearchStatus.Unregistered:
+      return (
+        <MetaContainer>
+          <Trans
+            defaults="Token info is needed before swapping this token. Please go to <a>Token Registry Center</a> to submit token info first."
+            components={{
+              // eslint-disable-next-line jsx-a11y/anchor-has-content
+              a: <a href="https://github.com/glias/token-list" target="_blank" rel="noreferrer noopener" />,
+            }}
+          />
+        </MetaContainer>
+      );
+    default:
+      return null;
+  }
+};
+
 export function GroupedAssetList<A extends Asset, K extends Key>(props: GroupedAssetListProps<A, K>) {
-  const { group, assets, onSelected } = props;
+  const { group, assets, onSelected, enableSearch } = props;
 
   const grouped = useMemo(() => groupBy(group, assets), [assets, group]);
+  const [searchValue, setSearchValue] = useState('');
```

Comment:
This does achieve the function of filter, but it does not feel very good to comply with the open-closed principle  , if we can decouple Search and List, this may leave the possibility of replacing the component later. Now gliaswap only supports Ethereum assets, but later it may support EOS or even BTC, so we will have to rewrite this component

```tsx
// add an property on the component
interface AssetSelectorProps {
  ...
  onSearch?: (userInput: string) => Promise<void>;
}

const AssetSelector = () => {
  // ...

  return (<>
   ...
  {onSearch && <Search onSearch={onSearch} />}
  </>)
}


// useSearchAsset.ts
function useSearchAsset () {
  // ...
}

// Example.tsx
function Example() {
  const { assets, onSearch } = useSearchAsset()

  return <AssetSelector onSearch={onSearch} assets={assets} />
}

```